libraries/Microsoft.Bot.Builder.Dialogs/WaterfallDialog.cs

```diff
@@ -105,6 +105,14 @@ public override async Task<DialogTurnResult> ResumeDialogAsync(DialogContext dc,
 
         protected virtual async Task<DialogTurnResult> OnStepAsync(WaterfallStepContext stepContext, CancellationToken cancellationToken)
         {
+            // Log Waterfall Step event. Each event has a distinct name to hook up
+            // to the Application Insights funnel.
+            var eventName = $"{Id}.Step{stepContext.Index + 1}of{_steps.Count}";
```

Comment:
It seems a little onerous to have to _name the events_ given that **{waterfall_dialog_name}/{waterfall_step_index}** is by definition a unique name at this scope (scope being defined by the containing dialogset.)

Could we at least make the name optional.

Reply:
Not sure I follow - if I have a dialog (`profileDialog`), and 4 waterfall steps, I'll get 4 events if someone makes it through:
`profileDialog.Step1of4`
`profileDialog.Step2of4`
`profileDialog.Step3of4`
`profileDialog.Step4of4`
The proposal @gabog has is to make this:
`profileDialog.MyFirstStepFunctionName` 
To make it clearer.. Have to make sure we can get function name in all languages..

Reply:
This should be all resolved now.  It takes the name of the step - for lambda's reverts back to naming above..